src/PostgREST/Error.hs

```diff
@@ -459,17 +460,26 @@ type Authenticated = Bool
 instance PgrstError PgError where
   status (PgError authed usageError) = pgErrorStatus authed usageError
 
-  headers (PgError _ (SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ResultError (SQL.ServerError "PGRST" m d _ _p))))) =
-    case parseRaisePGRST m d of
-      Right (_, r) -> map intoHeader (M.toList $ getHeaders r)
-      Left e       -> headers e
+  headers err@(PgError _ (SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ResultError (SQL.ServerError code m d _ _p))))) =
+    case code of
+      "57014" -> proxyStatusHeader (Just "57014") : statusBasedHeaders (status err)
```

Comment:
Hm, why just this code and not all the codes? The discussion on https://github.com/PostgREST/postgrest/issues/2967#issuecomment-2713224061 only mentioned `57014` as an example.

Reply:
Oh, okay. So it's for all `pg` errors. Got it.